### PR TITLE
Feed forward chunking others

### PR DESCRIPTION
**Pull request by Pradhy729**
Adding feed forward chunking to other models. Based on #6024

### CODE REVIEW

**Comment by patrickvonplaten** [comment]
https://github.com/huggingface/transformers/pull/6024 is merged :-) Great work @Pradhy729! It would be a good idea to rebase this PR to current master so that you can easily leverage the tests that were added in https://github.com/huggingface/transformers/pull/6024 just by setting the flag `test_chunking=True` for all models you want to add here.

**Comment by Pradhy729** [comment]
Yes - definitely will do. Was just waiting for the merge. Thanks for adding the tests.

**Comment by Pradhy729** [comment]
@patrickvonplaten Feed forward chunking has been added for the following:
1. Albert
2. Distillbert
3. Longformer
4. XLNet

+ changed model signature to have callable as first positional argument.

**Comment by Pradhy729** [comment]
Hi @patrickvonplaten --> Can you review and approve if this looks good?

File: src/transformers/configuration_utils.py
Comment (patrickvonplaten) [inline src/transformers/configuration_utils.py:191]: great, thanks for adding this!

File: src/transformers/modeling_albert.py
Comment (patrickvonplaten) [inline src/transformers/modeling_albert.py:311]: (nit) better name would arguable be `ffn_chunk`

File: src/transformers/modeling_utils.py
Comment (patrickvonplaten) [inline src/transformers/modeling_utils.py:1450]: Thanks for changing that. @LysandreJik - as you said this is the better order of the arguments and should be fine in terms of breaking backward compatibility

File: tests/test_modeling_common.py
Comment (patrickvonplaten) [inline tests/test_modeling_common.py:63]: Can you remove the `test_chunking=True` statement in other test files as well?

**Comment by patrickvonplaten** [comment]
Hey @Pradhy729 - this looks great! 
1) Can you add the docstrings for `chunk_size_feed_forward` as explained in the comment above and delete the corresponding config param in Reformer and the Reformer docstring (You can just cut & paste the Reformer docstring here)
2) Can you please remove the `test_chunking=True` statements in the model specific test files -> I think it's only in test_modeling_bert.py actually.
3) It would be awesome if you try to rebase the branch to master (`git fetch upstream master`, `git rebase upstream/master`).
If you have too many merge conflicts - then I'll do it :-)

**Comment by Pradhy729** [comment]
@patrickvonplaten 
Done. Please review and let me know if there's anything else.

File: src/transformers/configuration_utils.py
Comment (patrickvonplaten) [reply src/transformers/configuration_utils.py:191]: actually it's aleady done - never mind

**Comment by patrickvonplaten** [comment]
LGTM! @Pradhy729 - great work!

**Review by patrickvonplaten — APPROVED** [review]

**Review by sgugger — APPROVED** [review]
Great addition, thanks a lot!

**Comment by patrickvonplaten** [comment]
Merging! Good job @Pradhy729

**Pull request merged**
